src/client/keycode.cpp

```diff
@@ -345,23 +378,18 @@ const KeyPress RMBKey("KEY_RBUTTON");
 // A simple cache for quicker lookup
 static std::unordered_map<std::string, KeyPress> g_key_setting_cache;
 
-const KeyPress &getKeySetting(const char *settingname)
+const KeyPress &getKeySetting(const std::string &settingname)
 {
 	auto n = g_key_setting_cache.find(settingname);
 	if (n != g_key_setting_cache.end())
 		return n->second;
 
 	auto &ref = g_key_setting_cache[settingname];
-	ref = g_settings->get(settingname).c_str();
+	ref = std::string_view(g_settings->get(settingname));
```

Comment:
would show the intent clearer IMO
```suggestion
	ref = KeyPress(g_settings->get(settingname));
```